lettermgmt/models/res_letter.py

```diff
@@ -0,0 +1,156 @@
+# -*- encoding: utf-8 -*-
+##############################################################################
+#
+#    Parthiv Pate, Tech Receptives, Open Source For Ideas
+#    Copyright (C) 2009-Today Tech Receptives(http://techreceptives.com).
+#    Copyright (C) 2015 Therp BV <http://therp.nl>.
+#    All Rights Reserved
+#
+#
+#    This program is free software: you can redistribute it and/or modify
+#    it under the terms of the GNU General Public License as published by
+#    the Free Software Foundation, either version 3 of the License, or
+#    (at your option) any later version.
+#
+#    This program is distributed in the hope that it will be useful,
+#    but WITHOUT ANY WARRANTY; without even the implied warranty of
+#    MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+#    GNU General Public License for more details.
+#
+#    You should have received a copy of the GNU General Public License
+#    along with this program.  If not, see http://www.gnu.org/licenses/.
+#
+##############################################################################
+from openerp.osv import fields, orm
+
+
+class res_letter(orm.Model):
+    """A register class to log all movements regarding letters"""
+    _name = 'res.letter'
+    _description = "Log of Letter Movements"
+    _inherit = 'mail.thread'
+
+    def _get_number(self, cr, uid, context=None):
+        if context is None:
+            context = {}
+        sequence_pool = self.pool.get('ir.sequence')
+        move_type = context.get('move', 'in')
+        return sequence_pool.get(
+            cr, uid, '%s.letter' % move_type, context=context)
+
+    _columns = {
+        'name': fields.text('Subject', help="Subject of letter."),
+        'folder_id': fields.many2one(
+            'letter.folder', string='Folder',
+            help='Folder which contains letter.'),
+        'number': fields.char(
+            'Number', help="Auto Generated Number of letter.",
+            required=True),
+        'move': fields.selection(
+            [('in', 'IN'), ('out', 'OUT')], 'Move', readonly=True,
+            help="Incoming or Outgoing Letter."),
+        'type': fields.many2one(
+            'letter.type', 'Type',
+            help="Type of Letter, Depending upon size."),
+        'class': fields.many2one(
+            'letter.class', 'Class', help="Classification of Document."),
+        'date': fields.datetime('Letter Date', help='The letter\'s date'),
+        'snd_rec_date': fields.datetime(
+            'Sent / Received Date', required=True,
```

Comment:
This field is required now. I understand that the date received of an incoming letter can be required as it should be known in any state. But in the case of an outgoing letter, should it maybe be kept without a default and be set when setting the state to 'Sent' if it is still empty at that point?